demo-kotlin/app/src/main/java/com/example/placesdemo/PlaceAutocompleteActivity.kt

```diff
@@ -141,8 +141,8 @@ class PlaceAutocompleteActivity : AppCompatActivity() {
                 }
             }
             AutocompleteActivity.RESULT_ERROR -> {
-                val status = Autocomplete.getStatusFromIntent(intent)
-                binding.response.text = status.statusMessage
+                val status = Autocomplete.getStatusFromIntent(result.data)
+                binding.response.text = "Error: ${status.statusMessage}"
```

Comment:
This has a potential NPE.  How about the following change?

```suggestion
                binding.response.text = result.data?.let { intent ->
                    getString(
                        R.string.error_message,
                        Autocomplete.getStatusFromIntent(intent).statusMessage
                    )
                } ?: getString(R.string.error_unknown)
```


And add the following to demo-kotlin/app/src/main/res/values/strings.xml

```
<!-- message to show for errors received from the API. -->
  <string name="error_message" translatable="false">Error: %1$s</string>

  <!-- message to show if the error status is unavailable. -->
  <string name="error_unknown" translatable="false">Unexpected error</string>
```